cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -30,8 +31,9 @@ import (
 type PipelineAuthType string
 
 const (
-	AuthTypeFederated         PipelineAuthType = "federated"
-	AuthTypeClientCredentials PipelineAuthType = "client-credentials"
+	AuthTypeFederated               PipelineAuthType = "federated"
+	AuthTypeClientCredentials       PipelineAuthType = "client-credentials"
+	AzurePipelineClientIdEnvVarName string           = "AZURE_PIPELINE_CLIENT_ID"
```

Comment:
Can you add --principal-id to the pipeline config command?  I think that would make it easier for someone to find an ID and then use it versus name.

Reply:
I think we get the principal-id from ms-graph.
There would be an app-registration for the principal-id on each tenant where the principal-id is added. I think it is confusing for a customer to use ids directly...